File: app/src/main/java/data/vocab/models/SearchResult.java
package data.vocab.models;

import android.os.Parcelable;

import java.util.List;

import data.vocab.WordListEntry;

/**
 * Interface for the necessary components to return from a search.
 */
public interface SearchResult extends Parcelable {
    Vocabulary getVocabulary();
    List<WordListEntry> getRelatedWords();
}

File: app/src/main/java/data/vocab/model/SearchResult.kt
package data.vocab.model

import android.os.Parcelable

import data.vocab.shared.WordListEntry

/**
 * Interface for the necessary components to return from a search.
 */
interface SearchResult : Parcelable {
    val dictionaryEntry: DictionaryEntry
    val relatedWords: Array<WordListEntry>
}
